src/databricks/labs/ucx/contexts/workflow_task.py

```diff
@@ -109,3 +110,16 @@ def task_run_warning_recorder(self):
             self.inventory_database,
             int(self.named_parameters.get("attempt", "0")),
         )
+
+    @cached_property
+    def workflow_run_recorder(self):
+        return WorkflowRunRecorder(
+            self.sql_backend,
+            self.config.ucx_catalog,
+            workspace_id=int(self.named_parameters["workspace_id"]),
```

Comment:
```suggestion
            workspace_id=self.workspace_client.get_workspace_id(),
```
why do you need to pollute the whole named parameters stack just to add a parameter for one task?

Reply:
The reasoning was to avoid any API calls in the `Recorder`, see [this comment](https://github.com/databrickslabs/ucx/pull/2754#discussion_r1778109572)

Reply:
To avoid any API calls in the `WorkflowRunRecorder` makes it less likely to break